File: frontend/src/types/customer.ts
export interface Customer {
  id: number;
  name: string;
  email: string;
  created_at: string;
  completed_orders: number;
  completed_order_value: string;
}

export interface CustomerSummary {
  id: number;
  name: string;
  email: string;
  completed_orders: number;
  completed_order_value: string;
}

export interface CustomerList {
  items: CustomerSummary[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

File: backend/app/schemas/customer.py
from datetime import datetime
from decimal import Decimal
from typing import List
from pydantic import BaseModel, ConfigDict, Field


class CustomerResponse(BaseModel):
    id: int
    name: str
    email: str
    created_at: datetime
    completed_orders: int = 0
    completed_order_value: Decimal = Decimal("0.00")

    model_config = ConfigDict(from_attributes=True)


class CustomerSummaryResponse(BaseModel):
    id: int
    name: str
    email: str
    completed_orders: int
    completed_order_value: Decimal

    model_config = ConfigDict(from_attributes=True)


class CustomerListResponse(BaseModel):
    items: List[CustomerSummaryResponse]
    total: int
    page: int = 1
    page_size: int = 10
    total_pages: int = 1